### PR TITLE
fix: Various chart axis + checkpoint bugs [WEB-911]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

On the Trial Details page, LineChart was always displaying "Batches" in the X-Axis and tooltip. Adjusted to show `String(xAxis)` which covers Batches and Time.

Three other issues:
- Single point series (ex: a validation series with one record) were invisible unless also a checkpoint or hovered. Changed the drawPoints plugin to return `true` in this case, which shows an unfilled circle.
- checkpoint.endTime occurs after the metric is recorded, so the best checkpoint may match to a point on x = checkpoint.totalBatches but not on x = checkpoint.endTime (see experiment used in test).  If we're going to have separate series for batches and time then probably checkpoint(s) need to be sent to the chart as their own series?  But for now in this chart we can expect a batch # to be associated with every time, this is my use of timeHelpers.
- checkpoint.endTime offset also adds a new x-value which disrupts `dataForAxis` (which assumed training data has all x-values for clicking and tooltip) - replaced with xVals array per chart

## Test Plan

- Open `http://localhost:3000/det/experiments/1651/overview?f_chart=on`
- See one checkpoint diamond on accuracy and two on loss.  Hovering over each shows "best checkpoint"
- Switch x-axis to time. The accuracy checkpoint is still present and clickable. Both loss checkpoints are visible though spaced out.  The checkpoints in accuracy and loss plots have a "best checkpoint" tooltip and are clickable

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by mapmeld** [comment]
Changes to backend - on http://latest-master.determined.ai:8080/api/v1/trials/compare?trialIds=5248&maxDatapoints=1000&metricNames=accuracy&startBatches=0&metricType=METRIC_TYPE_VALIDATION&scale=SCALE_LINEAR you can see that we added time and batch metrics, but also doubled. This is because we appended to the metrics object in both functions -- it should be done once and we can do this outside of these fns

**Review by NicholasBlaskey — APPROVED** [review]
backend changes look good

File: webui/react/src/pages/TrialDetails/F_TrialDetailsOverview.tsx [hunk 1]
```diff
@@ -56,19 +56,30 @@ const TrialDetailsOverview: React.FC<Props> = ({ experiment, trial }: Props) =>
...
-    return checkpoint && xAxis === XAxisDomain.Batches
-      ? {
-          [checkpoint?.totalBatches]: checkpoint,
+    const timeHelpers: Record<XAxisVal, CheckpointWorkloadExtended> = {};
+    if (data && xAxis === XAxisDomain.Time && checkpoint?.totalBatches) {
+      Object.values(data).forEach((metric) => {
+        const matchIndex = metric.data[XAxisDomain.Batches]?.findIndex(
+          (pt) => pt[0] === checkpoint.totalBatches,
+        );
+        if (matchIndex !== undefined && matchIndex >= 0) {
```
Comment (julian-determined-ai) [inline webui/react/src/pages/TrialDetails/F_TrialDetailsOverview.tsx:68]: Just wanted to note that it seems like this logic will need to be updated when `epochs` are introduced but happy to defer that for now.

**Review by julian-determined-ai — APPROVED** [review]

File: webui/react/src/pages/TrialDetails/F_TrialDetailsOverview.tsx (hunk 1 above)
Comment (mapmeld) [reply webui/react/src/pages/TrialDetails/F_TrialDetailsOverview.tsx:68]: I agree; my idea while going over these is that LineChart will need to put checkpoints in a new series
